src/pages/ExerciseDetails/index.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import styles from './exerciseDetails.module.css';
+import {Head} from './Head';
+import { VariantCard } from './VariantCard';
+import { Carousel, CarouselItem } from 'components/Carousel';
+import { SideGallery } from './SideGallery';
+import { Footer } from 'components';
+import { useParams, useNavigate } from 'react-router-dom';
+import { getExerciseBase } from 'services';
+import { ExerciseBase } from 'components/Exercises/models/exerciseBase';
+import { useTranslation } from "react-i18next";
+import { getLanguageByShortName } from "services/language";
+import { useExerciseStateValue } from 'state';
+import { ENGLISH_LANGUAGE_ID } from 'utils/consts';
+import { ExerciseTranslation } from 'components/Exercises/models/exerciseTranslation';
+import { Language } from 'components/Exercises/models/language';
+
+export const ExerciseDetails = () => {
+    const [exerciseState, setExerciseState] = useState<ExerciseBase>();
+    const [currentUserLanguageState, setCurrentUserLanguageState] = useState<Language>();
+    const [state, dispatch] = useExerciseStateValue();
+    const params = useParams();
+    const exerciseID = params.exerciseID ? parseInt(params.exerciseID) : 0;
+    
+    // used to detect language from browser
+    const [t, i18n] = useTranslation();
+    
+
+    // to redirect to 404
+    const navigate = useNavigate();
+
+    const fetchedExercise = useCallback( async () => {
+        try {
+            const exerciseReceived = await getExerciseBase(exerciseID);
+            // get the exercise translation according to the user's language
+            const translatedExercise: ExerciseTranslation | undefined = exerciseState?.getTranslation(currentUserLanguageState);
+            // all this is so that I can have all the details so I can render
+            if (translatedExercise !== undefined) {
+                //replace the translations with the unique desired language to display
+                exerciseReceived.translations = [translatedExercise];
```

Comment:
Why overwrite the translations list? We can just keep all the translations there and simply show the one we are interested in like in OverviewCard (`const exerciseTranslation = exerciseBase.getTranslation(language);`)

Reply:
I overwrite because this page only displays 1 translation at a time. I also use `exerciseBase.getTranslation(language)`.  We can still keep the list and filter out the translation to display.